### PR TITLE
Define Array.isArray using a type predicate

### PR DESCRIPTION
**Pull request by mhegazy**
With type predicates in, here is a change to the lib to make it use that in isArray definition.

### CODE REVIEW

**Comment by vladima** [comment]
:+1:

**Comment by JsonFreeman** [comment]
:+1:

File: src/lib/core.d.ts [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ interface ArrayConstructor {
     (arrayLength?: number): any[];
     <T>(arrayLength: number): T[];
     <T>(...items: T[]): T[];
-    isArray(arg: any): boolean;
+    isArray(arg: any): arg is Array<any>;
```
Comment (CyrusNajmabadi) [inline src/lib/core.d.ts]: could we allow this to be generic?  i.e.

```ts
isArray<T>(arg: any): arg is Array<T>
```

?

Comment (JsonFreeman) [inline src/lib/core.d.ts]: We could, but we should add a separate overload for it. Otherwise, not passing the type argument gives you `Array<{}>`

Comment (mhegazy) [inline src/lib/core.d.ts]: so are you expecting this to work:
```ts
var arr: Array<number> | Array<string>;

if (Array.isArray<number>(arr)) {
    arr[0].toFixed(); // but what does that mean at runtime?
}
```

Comment (JsonFreeman) [inline src/lib/core.d.ts]: @mhegazy I think you're right. Specifying the type is claiming to know something that you don't.

**Pull request merged**